File: algorithms/data-structures/arrays/Sparse Arrays/sparse-arrays.js
// Sparse Arrays
// https://www.hackerrank.com/challenges/sparse-arrays/problem

function matchingStrings(strings, queries) {
  let result = [];

  for (let query of queries) {
    let sum = 0;
    for (let string of strings) {
      if (query === string) {
        sum += 1;
      }
    }
    result.push(sum);
  }
  return result;
}

File: algorithms/data-structures/arrays/Sparse Arrays/sparse-arrays.py
# Sparse Arrays
# https://www.hackerrank.com/challenges/sparse-arrays/problem


def matchingStrings(strings, queries):
  result = []

  for query in queries:
    sum = 0
    for string in strings:
      if (query == string):
        sum += 1
    result.append(sum)
    
  return result
